src/main/java/org/sonarsource/scanner/jenkins/pipeline/WaitForQualityGateStep.java

```diff
@@ -156,9 +155,9 @@ private boolean checkTaskCompleted() throws IOException, InterruptedException {
       log("SonarQube task '%s' status is '%s'", step.taskId, ceTask.getStatus());
       switch (ceTask.getStatus()) {
         case WsClient.CETask.STATUS_SUCCESS:
-          ProjectQualityGate qualityGate = wsClient.getQualityGateWithAnalysisId(ceTask.getAnalysisId());
-          log("SonarQube task '%s' completed. Quality gate is '%s'", step.taskId, qualityGate.getStatus());
-          getContext().onSuccess(new QGStatus(qualityGate.getStatus()));
+          String status = wsClient.requestQualityGateStatus(ceTask.getAnalysisId());
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Annotate the parameter with @javax.annotation.Nullable in method 'requestQualityGateStatus' declaration, or make sure that null can not be passed as argument. ([squid:S4449](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS4449))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.jenkins-ci.plugins%3Asonar&pullRequest=90&issues=AWIv-I6m_wH-bYHl3c0F&open=AWIv-I6m_wH-bYHl3c0F)

[ISSUE_KEY]: AWIv-I6m_wH-bYHl3c0F